framework/py/flwr/common/serde.py

```diff
@@ -461,28 +467,33 @@ def config_record_from_proto(record_proto: ProtoConfigRecord) -> ConfigRecord:
 
 def recorddict_to_proto(recorddict: RecordDict) -> ProtoRecordDict:
     """Serialize RecordDict to ProtoBuf."""
-    return ProtoRecordDict(
-        arrays={
-            k: array_record_to_proto(v) for k, v in recorddict.array_records.items()
-        },
-        metrics={
-            k: metric_record_to_proto(v) for k, v in recorddict.metric_records.items()
-        },
-        configs={
-            k: config_record_to_proto(v) for k, v in recorddict.config_records.items()
-        },
-    )
+    item_cls = ProtoRecordDict.Item
+    items: list[ProtoRecordDict.Item] = []
+    for k, v in recorddict.items():
+        if isinstance(v, ArrayRecord):
+            items += [item_cls(key=k, array_record=array_record_to_proto(v))]
+        elif isinstance(v, MetricRecord):
+            items += [item_cls(key=k, metric_record=metric_record_to_proto(v))]
+        elif isinstance(v, ConfigRecord):
+            items += [item_cls(key=k, config_record=config_record_to_proto(v))]
+        else:
+            raise ValueError(f"Unsupported record type: {type(v)}")
+    return ProtoRecordDict(items=items)
```

Comment:
Can we add a test for this and the following functions to check whether order is preserved?